keps/sig-api-machinery/2161-apiserver-default-labels/README.md

```diff
@@ -0,0 +1,487 @@
+
+Since this just an immutable field, we don't expect any such indicators to be of relevance.
+
+* **Are there any missing metrics that would be useful to have to improve observability 
+of this feature?**
+
+Since this just an immutable field, we don't expect any such indicators to be of relevance.
+
+
+### Dependencies
+
+* **Does this feature depend on any specific services running in the cluster?**
+  None
+
+### Scalability
+
+* **Will enabling / using this feature result in any new API calls?**
+
+No, because it is just mutating the incoming created namespaces at the APIServer level.
+
+* **Will enabling / using this feature result in introducing new API types?**
+
+No.
+
+* **Will enabling / using this feature result in any new calls to the cloud 
+provider?**:
+
+No
+
+* **Will enabling / using this feature result in increasing size or count of 
+the existing API objects?**
+
+Yes. Each namespace object will marginally increase in size to accommodate an extra label.
+
+* **Will enabling / using this feature result in increasing time taken by any 
+operations covered by [existing SLIs/SLOs]?**
+
+No
+
+* **Will enabling / using this feature result in non-negligible increase of 
+resource usage (CPU, RAM, disk, IO, ...) in any components?**
+
+No
+
+### Troubleshooting
+
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How does this feature react if the API server and/or etcd is unavailable?**
+
+* **What are other known failure modes?**
+  For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+
+* **What steps should be taken if SLOs are not being met to determine the problem?**
+
+[supported limits]: https://git.k8s.io/community//sig-scalability/configs-and-limits/thresholds.md
+[existing SLIs/SLOs]: https://git.k8s.io/community/sig-scalability/slos/slos.md#kubernetes-slisslos
+
+## Implementation History
+* 2020-11-22 - Creation of the issue and the KEP 
+* 2020-11-30 - https://github.com/kubernetes/kubernetes/pull/96968/files example PR implementing this
+
+## Drawbacks
+
+* "It's kind of a special hack. The fact that it isn't generalized or generalizable 
+makes it feel special." - @thockin
+
+## Alternatives
```

Comment:
Missing alternatives:
* add a virtual label. I.e., in memory only, don't store it on the object.
* Like a virtual label but go farther: add a syntax for label selectors that permit them to reference the fields in already in field selectors.

Reply:
ah clever :) , yeah adding those